docs/ProgrammingCryptol/highAssurance/HighAssurance.tex

```diff
@@ -174,6 +174,51 @@ \subsection{Capturing test vectors}
   sure the tweak to the above example (removal of dummy single formal
   parameter) works.}
 
+%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+\subsection{Generating test vectors}
+\label{sec:gentestvec}
+
+Cryptol has the ability to generate test vectors for simple function types using the \texttt{:dumptests} REPL command.
+When \texttt{:dumptests} is given a file to dump the output and a function in scope, it will randomly generate inputs for the function and run them for you.
+The output will contain a tab delimited file in which the first column contains the output, and each column after contains the inputs to the function.
+The command \texttt{:dumptests} uses the same settings as \texttt{:check} \ref{sec:quickcheck}, this means you can set the number of tests you want using the \texttt{:set tests=100} command.
+
+The following is an example using \href{https://github.com/GaloisInc/cryptol-specs/blob/7468d170f5e9e3913cd12e29c76d8c041ef99dd0/Primitive/Keyless/Hash/SHA2/Instantiations/SHA256.cry}{SHA2-256 implemntation} in \href{https://github.com/GaloisInc/cryptol-specs}{cryptol-specs}.
```

Comment:
+1, better than my suggestion of rewording the explanation

Reply:
Going to make a simple function, but did want to show how it can be used on instances of a polymporphic function as well as ones with non `Bit` outputs.

Reply:
Gotcha. Yeah, I think having more complex examples for practical settings is useful, but it's nice to build up to them. In particular, it's nice if the first example has
- a small function that the user can see & understand
- visually different types for input and output so the column labels are more obvious

Reply:
Added new example that still has the components from this that I wanted to show off.